### PR TITLE
Support updating alerts and fix bug in caching

### PR DESCRIPTION
**Pull request by aquarat**
- Removes some non-generic code
- Fixes a bug related to caching where when an alert is opened the returned ID was appended to `openAlerts` but in fact that ID is a request ID, not an alert ID 🤦 
- Adds a stub of axios to handle retrieving the content of an alert (so we can re-use portions of it)
- Adds a stub of axios to handle updating an alert's description (this enables us to update a URL in an alert)
- Export path was altered, but probably needs further work to make it logical
- Exposes a setter for axios to allow for more reliable/easier testing

Created related tasks:
- Set up some basic tests
- Setup CI

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export * as promises from './promises';
 export * as logging from './logging';
 export * from './types';
-export * as opsGenie from './opsgenie';
+export * from './opsgenie';
```
Comment (aquarat) [inline src/index.ts:4]: Not so sure about this, it makes imports downstream pretty, I should probably do it for everything/one namespace.

File: src/opsgenie.ts [hunk 2]
```diff
@@ -28,6 +25,20 @@ enum AlertsCachingStatus {
...
+
+/**
+ * A setter for the axios instance used by this library. Useful for reliably mocking out Axios for tests.
+ *
+ * @param axiosArgument
+ */
+export const setAxios = (axiosArgument: any) => {
+  axios = axiosArgument;
+};
+
```
Comment (vponline) [inline src/opsgenie.ts:41]: Nice :+1:

**Review by vponline — APPROVED** [review]
LGTM :rocket:

File: src/opsgenie.ts [hunk 3]
```diff
@@ -316,172 +339,60 @@ export const sendOpsGenieHeartbeat = async (heartBeatServiceName: string, opsGen
...
-      }
-      case Metric.API_BEACON_DEVIATION: {
-        debugLog(JSON.stringify(metric, null, 2));
-        if (!metric.value) {
-          console.debug('Metric value undefined', metric);
-          return;
-        }
+  promisedResults
+    .filter((result) => result.status === 'rejected')
+    .map((rejection) => log('Alert close promise rejected', 'ERROR', rejection));
```
[the remark this answers is not in the document]
Comment (aquarat) [reply src/opsgenie.ts:371]: ahhhhhhhh I see, yes you made a similar comment in Telemetry. I'll give your approach a go :)

[the remark this answers is not in the document]
Comment (aquarat) [reply src/opsgenie.ts:371]: Implementing the `reduce` approach highlighted the use of `return` to handle error scenarios which arguably should have been `throw`. The reduce approach probably has better type safety :)

[the remark this answers is not in the document]
Comment (aquarat) [reply src/opsgenie.ts:371]: Looking at this again this should have been a `forEach` or similar as it doesn't actually return anything 😬, whoops.

**Pull request merged**